src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/Parsers/ExpressionParser.cs

```diff
@@ -68,11 +68,29 @@ public Expression Parse(string[] resourceTypes, string key, string value)
         public IncludeExpression ParseInclude(string[] resourceTypes, string includeValue, bool isReversed, bool iterate, IEnumerable<string> allowedResourceTypesByScope)
         {
             var valueSpan = includeValue.AsSpan();
-            if (!TrySplit(SearchSplitChar, ref valueSpan, out ReadOnlySpan<char> originalType))
+
+            SearchParameterInfo refSearchParameter = null;
+            bool wildCard = false;
+
+            // check before split if this is a wildcard match
+            if (valueSpan.Equals("*".AsSpan(), StringComparison.InvariantCultureIgnoreCase))
+            {
+                refSearchParameter = null;
+                wildCard = true;
+            }
+
+            if (!TrySplit(SearchSplitChar, ref valueSpan, out ReadOnlySpan<char> originalType) && !wildCard)
             {
                 throw new InvalidSearchOperationException(isReversed ? Core.Resources.RevIncludeMissingType : Core.Resources.IncludeMissingType);
             }
 
+            // check after split if this is a wildcard match
+            if (valueSpan.Equals("*".AsSpan(), StringComparison.InvariantCultureIgnoreCase))
+            {
+                refSearchParameter = null;
+                wildCard = true;
+            }
```

Comment:
Then remove the after check
```suggestion
```

Reply:
The second check is still needed as valueSpan is passed by ref and the value updated to be just the '*' if the incoming string is something like:  Observation:* .  But I updated it to shortcircuit if wildcard has already been set to true.